src/transforms.js

```diff
@@ -121,29 +121,36 @@ export const setTextSelection = (position, dir = 1) => tr => {
 // ```
 export const safeInsert = (content, position) => tr => {
   const hasPosition = typeof position === 'number';
-  const $from = hasPosition ? tr.doc.resolve(position) : tr.selection.$from;
-  const { parent, depth } = $from;
+  const { $from } = tr.selection;
+  const $insertPos = hasPosition
+    ? tr.doc.resolve(position)
+    : isNodeSelection(tr.selection)
+      ? tr.doc.resolve($from.pos + 1)
+      : $from;
```

Comment:
This is getting interesting 😛

Reply:
yeah 😬well if its unreadable, I can break it down into two `consts`